src/IW/Sync/Search.hs

```diff
@@ -1,97 +1,139 @@
-       , fetchAllHaskellIssues
-       , fetchHaskellIssuesByLabels
+       ( searchHaskellReposByDate
+       , searchAllHaskellIssues
+       , searchHaskellIssuesByLabels
        , fromGitHubIssue
        , fromGitHubRepo
+       , liftGithubSearchToApp
        , parseUserData
        ) where
 
-import GitHub (SearchResult (..), URL (..))
-import GitHub.Endpoints.Search (searchRepos, searchIssues)
+import UnliftIO (MonadUnliftIO)
+import Data.Time (Day (..))
+import GitHub (SearchResult (..), URL (..), RateLimit (..), Limits, Paths, QueryString, executeRequest', limitsRemaining)
+import GitHub.Endpoints.RateLimit (rateLimit)
 
 import IW.App (WithError)
+import IW.App.Error (githubErrToAppErr, throwError)
 import IW.Core.Issue (Issue (..), Label (..))
 import IW.Core.Repo (Repo (..), RepoOwner (..), RepoName (..))
 import IW.Core.SqlArray (SqlArray (..))
-import IW.App.Error (githubErrToAppErr, throwError)
+import IW.Time (julianDayToIso)
 
 import qualified GitHub
 import qualified Data.Text as T
 import qualified Data.Vector as V
 
 
--- | Fetch all repositories built with the Haskell language.
-fetchAllHaskellRepos
-    :: ( MonadIO m
-       , WithError m
-       , WithLog env m
-       )
-    => m [GitHub.Repo]
-fetchAllHaskellRepos = liftGitHubSearchToApp searchHaskellRepos
-
--- | Convert a value of the @GitHub.Repo@ type to a value of our own @Repo@ type.
-fromGitHubRepo :: GitHub.Repo -> Repo
-fromGitHubRepo githubRepo = Repo
-    { repoOwner      = RepoOwner $ GitHub.untagName $ GitHub.simpleOwnerLogin $ GitHub.repoOwner githubRepo
-    , repoName       = RepoName  $ GitHub.untagName $ GitHub.repoName githubRepo
-    , repoDescr      = fromMaybe "" $ GitHub.repoDescription githubRepo
-    , repoCategories = SqlArray []
-    }
-
--- | Fetch all open issues with Haskell language and the labels passed in to the function.
-fetchHaskellIssuesByLabels
-    :: ( MonadIO m
-       , WithError m
-       , WithLog env m
-       )
-    => [Label]
-    -> m [GitHub.Issue]
-fetchHaskellIssuesByLabels = liftGitHubSearchToApp . searchHaskellIssuesByLabels
-
--- | Fetch all open issues with Haskell language.
-fetchAllHaskellIssues
-    :: ( MonadIO m
-       , WithError m
-       , WithLog env m
-       )
-    => m [GitHub.Issue]
-fetchAllHaskellIssues = fetchHaskellIssuesByLabels []
-
--- | Lift a github search action to work within our monad stack.
-liftGitHubSearchToApp
+-- | This function is for lifting a GitHub search action to the App monad.
+liftGithubSearchToApp
     :: forall a env m.
        ( MonadIO m
+       , MonadUnliftIO m
        , WithError m
        , WithLog env m
        , Typeable a
        )
     => IO (Either GitHub.Error (SearchResult a))
     -> m [a]
-liftGitHubSearchToApp githubSearch = liftIO githubSearch >>= \case
+liftGithubSearchToApp githubSearch' = do
+    searchLimit <- getSearchRateLimit
+    if limitsRemaining searchLimit > 0
+        then performSearch
+        else liftGithubSearchToApp githubSearch'
```

Comment:
Let's for now go with the simpler solution: just do `throwError` is we reached the limit. We will configure restart strategy separately